Feature/smhe 1432 opvoeren alle gasmeter sleutels bij inladen shipment file g meter

natasja-n opened the pull request:
https://jenkins.fdp.osgp.cloud/view/SMHE-cucumber/job/gxf-smhe-cucumber-build/53/

natasja-n committed:
SMHE-1432 opvoeren alle gasmeter sleutels bij inladen shipment file G-meter.
add g meter keys

Signed-off-by: Natasja Nortier <natasja.nortier@alliander.com>

natasja-n committed:
SMHE-1432 opvoeren alle gasmeter sleutels bij inladen shipment file G-meter
refactor, add unit test

Signed-off-by: Natasja Nortier <natasja.nortier@alliander.com>

natasja-n committed:
SMHE-1432 opvoeren alle gasmeter sleutels bij inladen shipment file G-meter
revert return list of e meter key types

Signed-off-by: Natasja Nortier <natasja.nortier@alliander.com>

kroesctrl commented on osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/application/services/InstallationService.java:
```diff
@@ -122,12 +120,55 @@ private List<SecurityKeyType> determineKeyTypesToStore(final SmartMeteringDevice
             new IllegalArgumentException(msg));
       }
-      return Arrays.asList(G_METER_MASTER);
+      return gMeterKeyTypes;
```
Should you not always all key types as done for e-meter

natasja-n replied:
right, all key types are returned now.

kroesctrl commented on osgp/platform/osgp-domain-core/src/main/java/org/opensmartgridplatform/domain/core/valueobjects/smartmetering/SmartMeteringDevice.java:
```diff
@@ -36,6 +36,10 @@ public class SmartMeteringDevice implements Serializable {
   private byte[] globalEncryptionUnicastKey;
 
   private byte[] authenticationKey;
+  private byte[] globalEncryptionBroadcastKey;
```
Where do you use globalEncryptionBroadcastKey?

natasja-n replied:
The unused key is removed.

kroesctrl commented on osgp/shared/osgp-dto/src/main/java/org/opensmartgridplatform/dto/valueobjects/smartmetering/SmartMeteringDeviceDto.java:
```diff
@@ -36,6 +36,10 @@ public class SmartMeteringDeviceDto implements Serializable {
   private byte[] globalEncryptionUnicastKey;
 
   private byte[] authenticationKey;
+  private byte[] globalEncryptionBroadcastKey;
```
Where do you use globalEncryptionBroadcastKey?

natasja-n replied:
The unused key is removed.

kroesctrl commented on osgp/shared/osgp-ws-smartmetering/src/main/resources/schemas/installation-ws-smartmetering.xsd:
```diff
@@ -332,6 +332,11 @@
         type="xsd:hexBinary" minOccurs="0" />
       <xsd:element name="Authentication_key" type="xsd:hexBinary"
         minOccurs="0" />
+      <xsd:element name="Global_encryption_broadcast_key" type="xsd:hexBinary" minOccurs="0" />
```
Where do you use Global_encryption_broadcast_key?

natasja-n replied:
The unused key is removed.

kroesctrl reviewed (changes requested):
I would expect there is also a check in a cucumber file like: AddMBusDevice.feature to check if keys exist after request

natasja-n committed:
SMHE-1432 opvoeren alle gasmeter sleutels bij inladen shipment file G-meter
add keys in AddMBusDevice.feature, return all gmeter keys in determineKeyTypesToStore, remove unused key

Signed-off-by: Natasja Nortier <natasja.nortier@alliander.com>
